### PR TITLE
feat: add ability to create user account from client

### PR DESCRIPTION
**Pull request by wreality**
User creation front end functionality #91.
* Update interface for Register form
* Validate data locally as much as possible before submitting
* Utilize createUser mutation for creating a user
* Unit test new components.

Reference #91 for acceptance criteria & #60

### CODE REVIEW

File: client/src/pages/Login.vue [hunk 1]
```diff
@@ -13,11 +13,11 @@
             square
             ref="username"
             v-model="form.username"
-            :label="$t('auth.login_fields.username')"
+            :label="$t('auth.fields.username')"
             @keypress.enter="$refs.password.focus()"
             autofocus
           >
-            <template v-slot:prepend>
+            <template #prepend>
```
Comment (gmeben) [inline client/src/pages/Login.vue:20]: Nothing that this is the only place in this file that uses the `v-slot` shorthand.

**Review by gmeben — APPROVED** [review]

File: client/src/pages/Login.vue (hunk 1 above)
Comment (wreality) [reply client/src/pages/Login.vue:20]: I'll fix the other one.  Admittedly I don't think I intended to include this change, but I do prefer the shorthand (which is probably why it got picked up... I probably did a find and replace on s/v-slot:/#/ )

**Comment by wreality** [comment]
@gmeben I've got a couple of little things that I noticed going back through the I'm going to fix before merging. Thanks for the review!

File: client/src/pages/Login.vue (hunk 1 above)
Comment (gmeben) [reply client/src/pages/Login.vue:20]: Yeah, I noticed you changed it to the shorthand elsewhere. Seems like making this stylistic change consistent throughout the codebase could be a separate pull request itself.

**Pull request merged**